crates/uv-resolver/src/lock.rs

```diff
@@ -255,11 +256,47 @@ impl Distribution {
                     let built_dist = BuiltDist::Registry(reg_dist);
                     Dist::Built(built_dist)
                 }
+                SourceKind::Path => {
+                    let filename: WheelFilename = wheel.filename.clone();
+                    let path_dist = PathBuiltDist {
+                        filename,
+                        url: VerbatimUrl::from_url(self.id.source.url.clone()),
+                        path: self.id.source.url.to_file_path().unwrap(),
+                    };
+                    let built_dist = BuiltDist::Path(path_dist);
+                    Dist::Built(built_dist)
+                }
+
                 // TODO: Handle other kinds of sources.
                 _ => todo!(),
             };
         }
-        // TODO: Handle source dists.
+
+        if let Some(sdist) = &self.sdist {
+            return match self.id.source.kind {
+                SourceKind::Path => {
+                    let path_dist = PathSourceDist {
+                        name: self.id.name.clone(),
+                        url: VerbatimUrl::from_url(self.id.source.url.clone()),
+                        path: self.id.source.url.to_file_path().unwrap(),
```

Comment:
It's a bit strange that we require both `url` and `path`, but I think the intent is that we do this fallible conversion when constructing to the type so that we don't have to do it everywhere that we _need_ the path.

Reply:
Yeah that makes sense I think.

(I am fond of the idea of encapsulating the distribution types and defining constructors for them. That way, you can add derived fields or whatever else without much fuss. But that's a big refactor and I am not altogether offended by pure data structs. They have their advantages...)